### ISSUE TITLE
MasterData client delivers infinite/repeated data on scroll and search methods

### ISSUE THREAD

**Issue by rolimans** [vtex/node-vtex-api#460]
When using client factories that use the MasterData client (such as `masterDataFor` available at [io-clients](https://github.com/vtex/io-clients)) the data returned from search and scroll methods is repeated and infinite. Such bug occurs because the http request made is cached and as the `search` and `scroll` queries are the same (without the query string) throughout the iterations the http client returns the cached result.

## Expected Behavior

When executing a scroll/search the results should change on a page number increase or in a scroll iteration.

## Current Behavior

When executing a scroll/search the results returned are the same regardless the page number or the current scroll iteration.

## Possible Solution

Setting the cache argument of the request to `CacheType.None` worked for me.

## Steps to Reproduce (for bugs)

1. Create a `masterDataFor` client in an IO app
2. Execute the code below

```
const records: T[] = []

    const lastResult: { hasNext: boolean; token: string | undefined } = {
      hasNext: true,
      token: undefined,
    }

    while (lastResult.hasNext) {
      const { data, mdToken } = await mdClient.scroll({
        size: 100,
        mdToken: lastResult.token,
        fields: ['_all'],
      })

      records.push(...(data as T[]))

      console.log(data[0].id)

      lastResult.hasNext = !!data.length
      if (!lastResult.token) {
        lastResult.token = mdToken
      }
    }
```
3. Notice how the while falls in an infinite loop printing always the same id from the first fetched document.

## Context

I had to switch from using the `masterDataFor` client to making raw http requests to the MasterData endpoint

## Your Environment

* Version used:
    "@vtex/api": "^6.44.0",
    "@vtex/clients": "^2.17.0",

**Comment by rolimans** [comment]
Submitted fix on #461

**Comment by rolimans** [comment]
Unfortunately the merge of #461 failed on llinting stage because of one property that was not in alphabetical order. Will fix and submit again. I'm sorry for the mistake, this is my first PR as part of the organization.

**Comment by rolimans** [comment]
Submitted new fix on #461

**Comment by rolimans** [comment]
Found a much better way to fix this, than setting the Cache to None in all requests, will assign this to me and fix as soon as I can.
